### DISCUSSION TITLE
"Bus Error" and "Segmentation Fault" errors / Unexpected Interrupt on CPU 0

### DISCUSSION THREAD

**Discussion by ExileIsland** [memtest86plus/memtest86plus#349] [Q&A]
Now I'm throwing things and nothing is sticking - so just tell me if I'm going down the wrong avenue of troubleshooting here... But I host some game servers on an HPE DL380 I have at the house, everything has been going fine and dandy until last night. There had been no major (or even minor) changes since the last reboot of the server and it was functioning. 

However, I believe we had a transformer blow down the street and our power was flickering on and off. I didn't have my UPS installed on this system at that moment. Before I was able to get a UPS actually installed on one of the redundant PSU's, power flickered at least two more times - causing total loss of power for the server.

Since this, I'm getting different errors while trying to run the dedicated server programs. I've gotten some "bus errors" (which I unfortunately don't have any further info for) on some launches, and some "segmentation fault" errors. So - while I don't know much about the system going super in depth like this, but I was reading that these errors *could* potentially spawn from memory issues. I was directed to use your tool, Memtest. Upon downloading the .iso and trying to boot to it (mounting the .iso over iLO) - I get the following blue screen once it boots up, stating: "Unexpected Interrupt on CPU 0, Type: Invalid Op" followed by what looks like a bunch of addresses. 

This is the screen cap of the Memtest86, I'm accessing it over iLO - so it is running on the base machine, not a VM. This is just the output of the server display.
![image](https://github.com/memtest86plus/memtest86plus/assets/125319372/ac06b0ad-1fa7-4eda-a846-3ef71bf0ff98)

and also, for the record, here is what the dedicated program error is looking like, not sure if it will be helpful at all here but thought I should include it for completeness.

"Caught fatal signal - signo:7 code:2 errno:0 addr:0x7f0f401b3c54
Obtained 13 stack frames.
#0  0x007f101425afd0 in __sigaction
#1  0x007f10103c69b7 in mono_metadata_type_hash
#2  0x007f10103dc9f9 in mono_g_hash_table_lookup_extended
#3  0x007f10103dc975 in mono_g_hash_table_lookup_extended
#4  0x007f10103dc94b in mono_g_hash_table_lookup
#5  0x007f101040ee3f in mono_type_get_object
#6  0x007f101040f128 in mono_type_get_object
#7  0x007f10103ac428 in mono_lookup_icall_symbol
#8  0x00000041f330ba in (wrapper managed-to-native) System.RuntimeTypeHandle:GetElementType (System.RuntimeType)
#9  0x00000041f321c4 in System.Collections.Generic.List`1<T_REF>:CopyTo (T_REF[],int)
#10 0x000000420918b4 in Newtonsoft.Json.Serialization.DefaultContractResolver:CreateProperty (System.Reflection.MemberInfo,Newtonsoft.Json.MemberSerialization)
#11 0x0000004208e9bb in Newtonsoft.Json.Serialization.DefaultContractResolver:CreateProperties (System.Type,Newtonsoft.Json.MemberSerialization)
#12 0x0000004208bc61 in Newtonsoft.Json.Serialization.DefaultContractResolver:CreateObjectContract (System.Type)"

**Comment by martinwhitaker** [comment]
Memtest86+ is loaded into and runs from system memory, so if the DRAM is too badly broken, it won't function properly. The same is true for any diagnostic program, unless it loads and runs from the non-volatile memory that contains the BIOS. And of course it could be your CPU that has been damaged, not the DRAM.

If you have more than one memory DIMM in the machine, try operating with just one DIMM installed at a time (assuming your BIOS allows that), to see if just one of the DIMMs is affected.

**Reply by ExileIsland** [reply]
Thank you for replying. It ended up being that a power outage seemed to have corrupted some of the essential files to operating the actual server files. The OS still functioned properly and after a full reinstall of the server, everything is working great. I was never able to get the memtest to run though. the iLO reports that the RAM is is good condition but I was still curious what the memtest would report.
Next time I take the server down for maintenance, I will try out your suggestion. But I don't think that anything is damaged in this way, as otherwise the system wouldn't be running like it is.

Is there any other causes that would make it throw this interrupt on the CPU like this? This is a dual CPU system, also 0 could that cause issues? Thanks in advance.
-Clay.
